src/thor/trace_route_action.cc

```diff
@@ -96,6 +96,13 @@ std::list<TripLeg> thor_worker_t::trace_route(valhalla_request_t& request) {
         auto map_match_results = map_match(request, controller);
         if (!map_match_results.empty()) {
           trip_paths = std::get<kTripLegIndex>(map_match_results.at(0));
+          if (trip_paths.empty())
+            throw std::exception{};
+          for (const auto& tp : trip_paths) {
+            if (tp.node().empty()) {
+              throw std::exception{};
```

Comment:
we need to look at the code history here, i dont see why we need a try catch around this, we may as well just throw 442 if any of these are empty